src/oInput/oInput.js

```diff
@@ -0,0 +1,98 @@
+import mapEvents from '../utils/eventMapper';
+import './input.css';
+import o from 'ojs-core';
+import generateRandomHash from '../utils/generateRandomHash';
+
+const TYPES = {
+    text: 'Text field',
+    date: 'Date field',
+    email: 'E-mail field',
+    month: 'Month field',
+    number: 'Number field',
+    password: 'Password field',
+    tel: 'Tel field',
+    file: 'File field',
+    url: 'URL field',
+};
+
+class oInput {
+    constructor(config) {
+        this.store = {
+            label: '',
+            type: 'text',
+            typeSpanText: false,
+            placeholder: '',
+            required: true,
+            index: false,
+            labelClass: 'input__label',
+            labelStyle: false,
+            inputClass: 'input__field',
+            inputStyle: false,
+            attributes: [],
+        };
+
+        this.configMerge(config);
+        this.store.id = config.id ? `${config.id}--input` : `${generateRandomHash()}--input`;
+    }
+
+    configMerge(config) {
+        Object.assign(this.store, config);
+        if (config.events) {
+            this.store.events = [...config.events, ...this.concatEvents(config)];
+        }
+    }
+
+    concatEvents(config) {
+        const defaultEvents = [
```

Comment:
To jest zawsze takie samo. Dobrze widzę? :D Jeśli tak to nie ma sensu tego trzymać w tej metodzie. Może to wylądować nawet poza ciałem klasy. Co za tym idzie istnienie tej metody trochę traciłoby sens.